CustomSecuritySample/README.md

```diff
@@ -0,0 +1,216 @@
+-	Locate the <AuthenticationTypes> element and modify the settings as follows: 
+	
+	```xml
+	<Authentication>
+		<AuthenticationTypes> 
+			<Custom/>
+		</AuthenticationTypes>
+		<RSWindowsExtendedProtectionLevel>Off</RSWindowsExtendedProtectionLevel>
+		<RSWindowsExtendedProtectionScenario>Proxy</RSWindowsExtendedProtectionScenario>
+		<EnableAuthPersistence>true</EnableAuthPersistence>
+	</Authentication>
+	```
+
+-	Locate the <Security> and <Authentication> elements, within the <Extensions> element, and modify the settings as follows: 
+
+	```xml
+	<Security>
+		<Extension Name="Forms" Type="Microsoft.Samples.ReportingServices.CustomSecurity.Authorization, Microsoft.Samples.ReportingServices.CustomSecurity" >
+		<Configuration>
+			<AdminConfiguration>
+				<UserName>username</UserName>
+			</AdminConfiguration>
+		</Configuration>
+		</Extension>
+	</Security>
+	```
+	```xml
+	<Authentication>
+		<Extension Name="Forms" Type="Microsoft.Samples.ReportingServices.CustomSecurity.AuthenticationExtension,Microsoft.Samples.ReportingServices.CustomSecurity" />
+	</Authentication> 
+	```
+
+-	Locate the <UI> element and update it as follows: 
+	
+	```xml
+	<UI>
+		<CustomAuthenticationUI>
+			<loginUrl>/Pages/UILogon.aspx</loginUrl>
+			<UseSSL>True</UseSSL>
+		</CustomAuthenticationUI>
+		<ReportServerUrl>http://<server>/ReportServer</ReportServerUrl>
+		<PageCountMode>Estimate</PageCountMode>
+	</UI>
+	```
+	
+Note: 
+If you are running the sample security extension in a development environment that does not have a Secure Sockets Layer (SSL) certificate installed, you must change the value of the <UseSSL> element to False in the previous configuration entry. We recommend that you always use SSL when combining Reporting Services with Forms Authentication. 
+
+To modify the RSSrvPolicy.config file 
+-	You will need to add a code group for your custom security extension that grants FullTrust permission for your extension. You do this by adding the code group to the RSSrvPolicy.config file.
+-	Open the RSSrvPolicy.config file located in the <install>\ReportServer directory. 
+-	Add the following <CodeGroup> element after the existing code group in the security policy file that has a URL membership of $CodeGen as indicated below and then add an entry as follows to RSSrvPolicy.config. Make sure to change the below path according to your ReportServer installation directory:
+	
+	```xml
+	< CodeGroup
+		class="UnionCodeGroup"
+		version="1"
+		Name="SecurityExtensionCodeGroup" 
+		Description="Code group for the sample security extension"
+		PermissionSetName="FullTrust">
+	< IMembershipCondition 
+		class="UrlMembershipCondition"
+		version="1"
+		Url="C:\Program Files\Microsoft SQL Server\MSRS11.MSSQLSERVER\Reporting Services\ReportServer\bin\Microsoft.Samples.ReportingServices.CustomSecurity.dll"/>
```

Comment:
In the document I have mentioned that we are assuming that RS has been installed to the default location as C:\Program Files\Microsoft SQL Server\MSRS13.MSSQLSERVER (considering that RS has been installed under default instance of SQL i.e. MSSQLSERVER.).
Do this still needs to be changed?